third_party/nvfuser/csrc/ir_utils.cpp

```diff
@@ -465,6 +477,14 @@ class ValReplacementMutator : private OptOutMutator {
     for (auto stmt : more_stmts) {
       mutate(stmt);
     }
+    // Original update order: input_tv, expr(including attribute), output_tv. In
```

Comment:
This seems like a general problem with mutating `ScatterOp`, so shouldn't it be addressed in `OptOutMutator`?